user_guide.adoc

```diff
@@ -373,18 +373,18 @@ strings, but the same features should work on any user types as well.
 
 Another popular way to avoid allocating huge fixed-size buffers is to
 use some sort of trailing delimiter. The most well-known example of
-this is probably null-terminated string which became a standard of
-strings representation in C:
+this is probably the null-terminated string which became a standard
+string representation in C:
 
 ....
 61 62 63 00
 ....
 
-This 4 bytes actually represent 3-character string "abc", and extra
-one trailing byte "0" (AKA null) serves as a delimiter or
-terminator. By agreement, C strings cannot include 0 byte: every time
+These 4 bytes actually represent the 3-character string "abc", plus one extra
+trailing byte "0" (AKA null) which serves as a delimiter or
+terminator. By agreement, C strings cannot include a 0 byte: every time
```

Comment:
```suggestion
terminator. By agreement, C strings cannot include a byte 0: every time
```

Reply:
Not sure about this. I have a feeling (although haven't web searched to check), that people often refer to the "zero byte" in the context of string terminators.

Reply:
> Not sure about this. I have a feeling (although haven't web searched to check), that people often refer to the "zero byte" in the context of string terminators.

Yeah, you're probably right, this was just my gut feeling, as I have never seen "0 byte" where 0 is written as a numeral, I'd say it's quite unusual. Maybe "zero byte" would be better?

Reply:
Yep, I've changed to "zero byte".